new-lamassu-admin/src/pages/Maintenance/Wizard/Wizard.js

```diff
@@ -0,0 +1,103 @@
+import * as R from 'ramda'
+import React, { useState } from 'react'
+import * as Yup from 'yup'
+
+import Modal from 'src/components/Modal'
+
+import WizardSplash from './WizardSplash'
+import WizardStep from './WizardStep'
+
+const MODAL_WIDTH = 554
+const MODAL_HEIGHT = 520
+const CASHBOX_DEFAULT_CAPACITY = 500
+
+const Wizard = ({ machine, cashoutSettings, locale, onClose, save, error }) => {
+  const [{ step, config }, setState] = useState({
+    step: 0,
+    config: { active: true }
+  })
+
+  const numberOfCassettes = machine.numberOfCassettes
+  const numberOfCassetteSteps = R.isEmpty(cashoutSettings)
+    ? 0
+    : numberOfCassettes
+  const LAST_STEP = numberOfCassetteSteps + 1
+
+  const title = `Update counts`
+  const isLastStep = step === LAST_STEP
+
+  const onContinue = it => {
+    const cashbox = config?.wasCashboxEmptied === 'YES' ? 0 : machine?.cashbox
+
+    console.log('Wizard.js:Wizard:it', it)
+    if (isLastStep) {
+      save(
+        machine.id,
+        parseInt(cashbox),
+        parseInt(it[1] ?? 0),
+        parseInt(it[2] ?? 0),
+        parseInt(it[3] ?? 0),
+        parseInt(it[4] ?? 0)
+      )
```

Comment:
Not really a fan of this, would prefer if this transformation happened on the schema/validation side. Doesn't the cashout setup wizard solve this issue? Essentially what it does is this. Initial values are at `''`, similar to what is currently on this and the transformNumber is a function imported which works the same way as the `parseInt(it[X] ?? 0)`

```
const schema = () =>
    Yup.object().shape({
      cassette1: Yup.number().required(),
      cassette2:
        machine.numberOfCassettes > 1 && step >= 2
          ? Yup.number().required()
          : Yup.number()
              .transform(transformNumber)
              .nullable(),
      cassette3:
        machine.numberOfCassettes > 2 && step >= 3
          ? Yup.number().required()
          : Yup.number()
              .transform(transformNumber)
              .nullable(),
      cassette4:
        machine.numberOfCassettes > 3 && step >= 4
          ? Yup.number().required()
          : Yup.number()
              .transform(transformNumber)
              .nullable()
    })
```

Reply:
This was a pain to get "right"... `transform()` doesn't seem to do what we think it does.